### PR TITLE
Presence

### PR DESCRIPTION
**Pull request by moozzyk**
Continuation of https://github.com/aspnet/SignalR/pull/391 which was closed by accident.

### CODE REVIEW

**Comment by mikaelm12** [comment]
Looks like you need to rebase

**Comment by moozzyk** [comment]
Rebased on latest dev

File: samples/ChatSample/HubWithPresence.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public class HubWithPresence<TClient> : Hub<TClient>
+    {
+        private IUserTracker<HubWithPresence<TClient>> _userTracker;
+
+        public HubWithPresence(IUserTracker<HubWithPresence<TClient>> userTracker)
+        {
+            _userTracker = userTracker;
+        }
+
+        public Task<IEnumerable<UserDetails>> UsersOnline
```
Comment (davidfowl) [inline samples/ChatSample/HubWithPresence.cs]: Async properties are super weird

Comment (moozzyk) [inline samples/ChatSample/HubWithPresence.cs]: 👍

**Pull request merged**